saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/metadata/OpenSamlMetadataResolver.java

```diff
@@ -75,9 +79,21 @@ public String resolve(RelyingPartyRegistration relyingPartyRegistration) {
 		entityDescriptor.setEntityID(relyingPartyRegistration.getEntityId());
 		SPSSODescriptor spSsoDescriptor = buildSpSsoDescriptor(relyingPartyRegistration);
 		entityDescriptor.getRoleDescriptors(SPSSODescriptor.DEFAULT_ELEMENT_NAME).add(spSsoDescriptor);
+		this.parametersConsumer
+				.accept(new EntityDescriptorParameters(entityDescriptor, relyingPartyRegistration, spSsoDescriptor));
 		return serialize(entityDescriptor);
 	}
 
+	/**
+	 * Set a {@link Consumer} for modifying the OpenSAML {@link EntityDescriptor}
+	 * @param parametersConsumer a consumer that accepts an
+	 * {@link EntityDescriptorParameters}
+	 */
+	public void setParametersConsumer(Consumer<EntityDescriptorParameters> parametersConsumer) {
```

Comment:
Please add a `@since 5.7` tag.

Also, I think `setEntityDescriptorCustomizer` is a little bit more descriptive, similar to what is done in `OpenSaml4AuthenticationRequestResolver`.

Reply:
Should the EntityDescriptorParameters also have the `@Since` tag?